internal/gittuf-git/push.go

```diff
@@ -0,0 +1,32 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/gittuf/gittuf/internal/cmd/rsl"
+)
+
+func splitPushArgs(args []string) (rslRecordArgs []string) {
+	rslRecordArgs = []string{}
+
+	for _, arg := range args {
+		if !strings.HasPrefix(arg, "-") {
+			rslRecordArgs = append(rslRecordArgs, arg)
+		}
+	}
+	return rslRecordArgs
+}
+
+func gittufPush(gitArgs []string) {
+	rslRecordArgs := splitPushArgs(gitArgs[1:])
+	recordCmd := rsl.New()
+	rslArg := []string{}
+	for _, arg := range rslRecordArgs {
+		rslArg = append(rslArg, arg)
+		if err := recordCmd.RunE(nil, rslArg); err != nil {
```

Comment:
Now that https://github.com/gittuf/gittuf/pull/369 has been merged, we can run `gittuf rsl skip-rewritten` when recording in the RSL if the ref's state has changed, i.e., rebasing.